src/optimization/Optimizer.cpp

```diff
@@ -148,9 +148,11 @@ const OptimizationPass optimizations::COMBINE = OptimizationPass("CombineALUIins
 const OptimizationPass optimizations::UNROLL_WORK_GROUPS = OptimizationPass("UnrollWorkGroups", unrollWorkGroups, 170);
 const OptimizationPass optimizations::ADD_START_STOP_SEGMENT = OptimizationPass("AddStartStopSegment", addStartStopSegment, 180);
 const OptimizationPass optimizations::EXTEND_BRANCHES = OptimizationPass("ExtendBranches", extendBranches, 190);
+// Following index doesn't be considered.
```

Comment:
The index is considered, it is the sorting of the optimization-steps.
You should put this step somewhere before `VectorizeLoops` and `UnrollWorkGroups`.